mautic/mautic
# Added Recurring Campaign Functionality

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? |  Yes
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | #1606 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR will add Recurring campaign functionality. there will be a new button in campaign details page. if this button set to yes leads that are manually removed from campaign (via remove from this campaign in campaign builder) can reEnter the campaign if there are no scheduled events remaining.

#### Steps to test this PR:
1. Use the migration to add the new column
2. Clear cache
3. Create a new campaign
4. Set recurring to yes
5. In campaign builder make sure you remove some Contacts from campaign by selecting Remove from this campaign
6. Run the cron jobs.
7. Run the cron jobs again, this time Manually Removed contacts will reenter the campaign.

Special thanks to @mqueme for helping and @mazzim for the idea.
